src/v/transform/rpc/client.h

```diff
@@ -256,11 +260,12 @@ class client {
     model::node_id _self;
     std::unique_ptr<cluster_members_cache> _cluster_members;
     // need partition_leaders_table to know which node owns the partitions
-    std::unique_ptr<partition_leader_cache> _leaders;
-    std::unique_ptr<topic_metadata_cache> _topic_metadata;
-    std::unique_ptr<topic_creator> _topic_creator;
+    std::unique_ptr<kafka::data::rpc::partition_leader_cache> _leaders;
+    std::unique_ptr<kafka::data::rpc::topic_metadata_cache> _topic_metadata;
     ss::sharded<::rpc::connection_cache>* _connections;
     ss::sharded<local_service>* _local_service;
+    ss::sharded<kafka::data::rpc::client>* _kafka_client;
```

Comment:
It's more a `_data_client` because it's not kafka protocol, but kafka semantics. But that name sucks, so this is fine :)

Reply:
> that name sucks, so this is fine

literally cackled 😂 

but yeah agreed kafka_client is not excellent.